tests/unit/test_connection.py

```diff
@@ -0,0 +1,27 @@
+from dbt.adapters.dremio.api.rest.error import DremioRequestTimeoutException
+import pytest
+from unittest.mock import patch
+from dbt.adapters.dremio.connections import DremioConnectionManager
+from requests.exceptions import HTTPError
+from dbt.exceptions import FailedToConnectException
+
+
+class TestRetryConnections:
+    @patch("dbt.adapters.dremio.api.rest.endpoints._post")
+    @patch("dbt.contracts.connection.Connection")
+    # When you nest patch decorators the mocks are passed in to the decorated function in bottom up order.
+    def test_run_connection_retry_test(
+        self,
+        mocked_Connection,
```

Comment:
Do we want a capital C? I think I know why you did it, but not sure if that is necessary.

Reply:
I was just trying to distinguish between mocked classes and mocked functions. We can remove if you'd like.

Reply:
Change it to `mocked_connection_obj`.